src/Ztm.WebApi/TransactionConfirmationWatcher.cs

```diff
@@ -408,12 +387,12 @@ async Task<IEnumerable<Rule>> ITransactionConfirmationWatcherHandler<Rule>.Creat
 
         async Task IWatcherHandler<TransactionWatch<Rule>, Rule>.RemoveWatchAsync(TransactionWatch<Rule> watch, WatchRemoveReason reason, CancellationToken cancellationToken)
         {
-            if (reason == WatchRemoveReason.BlockRemoved)
+            if (reason.HasFlag(WatchRemoveReason.BlockRemoved))
             {
                 await this.watchRepository.UpdateStatusAsync(watch.Id, TransactionConfirmationWatchingWatchStatus.Rejected, cancellationToken);
                 await SetupTimerAsync(watch.Context);
             }
-            else
+            else if (reason.HasFlag(WatchRemoveReason.Completed))
```

Comment:
`else` is not correct here. Also we need to check for completed before block removed too.